src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -50,14 +51,25 @@ public async Task<string> GetSecretValue(Uri secretUri, string key, Cancellation
             else if (client != null)
             {
                 KeyVaultSecret secret;
-                secret = await client.GetSecretAsync(secretName, secretVersion, cancellationToken).ConfigureAwait(false);
-                secretValue = secret?.Value;
-                SetSecretInCache(key, secretValue);
+                bool success = false;
+
+                try
+                {
+                    secret = await client.GetSecretAsync(secretName, secretVersion, cancellationToken).ConfigureAwait(false);
+                    secretValue = secret?.Value;
+                    cachedSecret = new CachedKeyVaultSecret(secretValue);
+                    success = true;
+                }
+                finally
+                {
+                    SetSecretInCache(key, cachedSecret, success);
+                }
             }
             else if (_keyVaultOptions.SecretResolver != null)
             {
                 secretValue = await _keyVaultOptions.SecretResolver(secretUri).ConfigureAwait(false);
```

Comment:
What if a customer secret resolver throws? I guess we don't to want to call it like a crazy either.